ext/opentelemetry-ext-pymongo/src/opentelemetry/ext/pymongo/__init__.py

```diff
@@ -50,27 +52,51 @@
 COMMAND_ATTRIBUTES = ["filter", "sort", "skip", "limit", "pipeline"]
 
 
-def trace_integration(tracer_provider=None):
-    """Integrate with pymongo to trace it using event listener.
-       https://api.mongodb.com/python/current/api/pymongo/monitoring.html
+class PymongoInstrumentor(BaseInstrumentor):
+    _commandtracer_instance = None
+    # The instrumentation for PyMongo is based on the event listener interface
+    # https://api.mongodb.com/python/current/api/pymongo/monitoring.html.
+    # This interface only allows to register listeners and does not provide
+    # an unregister API. In order to provide a mechanishm to disable
+    # instrumentation an enabled flag is implemented in CommandTracer,
+    # it's checked in the different listeners.
 
-    Args:
-        tracer_provider: The `TracerProvider` to use. If none is passed the
-            current configured one is used.
-    """
+    def _instrument(self, **kwargs):
+        """Integrate with pymongo to trace it using event listener.
+        https://api.mongodb.com/python/current/api/pymongo/monitoring.html
 
-    tracer = get_tracer(__name__, __version__, tracer_provider)
+        Args:
+            tracer_provider: The `TracerProvider` to use. If none is passed the
+                current configured one is used.
+        """
 
-    monitoring.register(CommandTracer(tracer))
+        tracer_provider = kwargs.get("tracer_provider")
+
+        # Create and register a CommandTracer only the first time
+        if self._commandtracer_instance is None:
+            tracer = get_tracer(__name__, __version__, tracer_provider)
+
+            self._commandtracer_instance = CommandTracer(tracer)
+            monitoring.register(self._commandtracer_instance)
+
+        # If already created, just enable it
+        self._commandtracer_instance.enable = True
+
+    def _uninstrument(self, **kwargs):
+        if self._commandtracer_instance is not None:
+            self._commandtracer_instance.enable = False
 
 
 class CommandTracer(monitoring.CommandListener):
     def __init__(self, tracer):
         self._tracer = tracer
         self._span_dict = {}
+        self.enable = True
```

Comment:
```suggestion
        self.enabled = True
```

Reply:
will the above code have to change at the same time to reference the new name?

I guess if this is a nitpick, I'd vote "is_enabled": a little semantic notation indicating the value is a bool.

Reply:
Updated to `is_enabled` in https://github.com/open-telemetry/opentelemetry-python/pull/612/commits/4b020c778d49f17152408f7a2c8c777c5eb13b21.